crypto/evp/m_sigver.c

```diff
@@ -158,7 +158,7 @@ int EVP_DigestSign(EVP_MD_CTX *ctx, unsigned char *sigret, size_t *siglen,
 {
     if (ctx->pctx->pmeth->digestsign != NULL)
         return ctx->pctx->pmeth->digestsign(ctx, sigret, siglen, tbs, tbslen);
-    if (sigret != NULL && EVP_DigestSignUpdate(ctx, tbs, tbslen) <= 0)
```

Comment:
No. If `sigret == NULL`, then it's expected that the caller tries to figure out what the signature length will be. In that case, the `EVP_DigestSignUpdate` call is a waste of cycles (and I have no idea what corner cases it might hit as well)